FrameworkSystem/DB/InstalledComponentsDB.py

```diff
@@ -0,0 +1,758 @@
+  InstalledComponentsDB database
+  """
+
+  __tablename__ = 'InstalledComponents'
+  __table_args__ = {
+    'mysql_engine': 'InnoDB',
+    'mysql_charset': 'utf8'
+  }
+
+  componentID = Column( 'ComponentID', 
+                        Integer,
+                        ForeignKey( 'Components.ComponentID' ),
+                        primary_key = True )
+  hostID = Column( 'HostID',
+                    Integer,
+                    ForeignKey( 'Hosts.HostID' ),
+                    primary_key = True )
+  instance = Column( 'Instance',
+                      String( 32 ),
+                      primary_key = True )
+  installationTime = Column( 'InstallationTime',
+                              DateTime,
+                              primary_key = True )
+  unInstallationTime = Column( 'UnInstallationTime',
+                                DateTime )
+  installationComponent = relationship( 'Component',
+                                        backref = 'installationList' )
+
+  def __init__( self, instance = null(),
+                      installationTime = null(),
+                      unInstallationTime = null() ):
+    self.instance = instance
+    self.installationTime = installationTime
+    self.unInstallationTime = unInstallationTime
+
+  def fromDict( self, dictionary ):
+    """
+    Fill the fields of the InstalledComponent object from a dictionary
+    The dictionary may contain the keys: ComponentID, HostID, Instance,
+    InstallationTime, UnInstallationTime
+    """
+
+    self.componentID = dictionary.get( 'ComponentID', self.componentID )
+    self.hostID = dictionary.get( 'HostID', self.hostID )
+    self.instance = dictionary.get( 'Instance', self.instance )
+    self.installationTime = dictionary.get( 'InstallationTime',
+                                            self.installationTime )
+    self.unInstallationTime = dictionary.get( 'UnInstallationTime',
+                                              self.unInstallationTime )
+
+    return S_OK( 'Successfully read from dictionary' )
+
+  def toDict( self, includeComponents = False, includeHosts = False ):
+    """
+    Return the object as a dictionary
+    If includeComponents is True, information about which Components where
+    installed is included
+    If includeHosts is True, information about the Hosts where the
+    installations are is included
+    """
+
+    dictionary = {
+                  'Instance': self.instance,
+                  'InstallationTime': self.installationTime,
+                  'UnInstallationTime': self.unInstallationTime
+                  }
+
+    if includeComponents:
+      dictionary[ 'Component' ] = self.installationComponent.toDict()[ 'Value' ]
+    else:
+      dictionary[ 'ComponentID' ] = self.componentID
+
+    if includeHosts:
+      dictionary[ 'Host' ] = self.installationHost.toDict()[ 'Value' ]
+    else:
+      dictionary[ 'HostID' ] = self.hostID
+
+    return S_OK( dictionary )
+
+class InstalledComponentsDB( object ):
```

Comment:
The DB initialization includes look up of the DB parameters in the CS. This is the operation the same as in the DIRAC "classical" DB base class. This can be made a utility common to both implementations.